### PR TITLE
Add more logging around DocTableInfo construction

### PR DESCRIPTION
**Pull request by romseygeek**

### CODE REVIEW

**Comment by romseygeek** [comment]
This is still not failing at all reproducibly, either locally or on the Jenkins machines, so let's add a whole bunch of extra logging to see if we can see what the root cause is when we do get a failure.

**Comment by seut** [comment]
This contains some other changes than just logging, mistake?

**Comment by romseygeek** [comment]
There's a script to aid running a test multiple times, and a small refactoring that allows us to log mapping changes - I don't think there's anything else?

**Comment by romseygeek** [comment]
I've removed the beasting script as that's not really related

**Review by mfussenegger — COMMENTED** [review]
I'm not sure these are the kind of logging changes we want to keep indefinitely. 

Could we just keep this on this branch and run the tests somewhere in an endless loop?

**Comment by romseygeek** [comment]
> Could we just keep this on this branch and run the tests somewhere in an endless loop?

I've been doing this yesterday and today, and frustratingly have got no errors.  Quite why it fails once every 50 or so runs on Jenkins CI but never fails when run directly from bash I don't know.  But it seems to be totally unreproducible, hence why I'd like to add this extra logging.

File: server/src/main/java/io/crate/metadata/doc/DocSchemaInfo.java [hunk 1]
```diff
@@ -270,10 +275,12 @@ public void update(ClusterChangedEvent event) {
 
             IndexMetadata newIndexMetadata = newMetadata.index(indexName);
             if (newIndexMetadata == null) {
+                LOGGER.debug("Invalidating DocTableInfo for deleted table [{}] [{}]", tableName, newMetadata.indices().keys());
```
Comment (seut) [inline server/src/main/java/io/crate/metadata/doc/DocSchemaInfo.java:278]: Avoid `keys()` computation if debug isn't enabled.
```suggestion
                if (LOGGER.isDebugEnabled()) {
                    LOGGER.debug("Invalidating DocTableInfo for deleted table [{}] [{}]", tableName, newMetadata.indices().keys());
                }
```

File: server/src/main/java/io/crate/metadata/doc/DocTableInfo.java [hunk 2]
```diff
@@ -344,6 +348,7 @@ public Set<Reference> droppedColumns() {
      */
     public TranslogIndexer getTranslogIndexer() {
         if (this.translogIndexer == null) {
+            LOGGER.debug(() -> "Building translog indexer for " + this.ident + " using mapping " + this.mapping().mapping);
```
Comment (seut) [inline server/src/main/java/io/crate/metadata/doc/DocTableInfo.java:351]: Same here, please surround with `if (LOGGER.isDebugEnabled())` to void expensive operations if not enabled.

**Comment by romseygeek** [comment]
I tracked down this issue without having to add the logging in the end - due to a race between new index shards being opened via IndicesClusterStateService (updated with ClusterStateAppliers) and new DocTableInfos being created in Schemas (updated with ClusterStateListeners).  Fixed instead by #16322

**Pull request closed without merging**